### PR TITLE
Various fixes to correctness of glTF generation.

### PR DESCRIPTION
**Pull request by kring**
This is a PR into #842 so merge that first.

There are quite a few small fixes here. See the changelog for the full list.

### CODE REVIEW

File: Cesium3DTilesContent/test/TestPntsToGltfConverter.cpp [hunk 1]
```diff
@@ -238,6 +249,8 @@ TEST_CASE("Converts point cloud with RGBA to glTF") {
   Model& gltf = *result.model;
 
   CHECK(gltf.hasExtension<CesiumGltf::ExtensionCesiumRTC>());
+  CHECK(gltf.isExtensionUsed(ExtensionCesiumRTC::ExtensionName));
+  CHECK(gltf.isExtensionRequired(ExtensionCesiumRTC::ExtensionName));
   CHECK(gltf.nodes.size() == 1);
```
Comment (j9liu) [inline Cesium3DTilesContent/test/TestPntsToGltfConverter.cpp:254]: Doesn't hurt to add an extra check for the scene, too. (I can do that)

```suggestion
  CHECK(gltf.scenes.size() == 1);
  CHECK(gltf.nodes.size() == 1);
```

File: Cesium3DTilesContent/test/TestUpgradeBatchTableToExtStructuralMetadata.cpp [hunk 2]
```diff
@@ -2319,6 +2368,97 @@ TEST_CASE("Converts \"Feature Classes\" 3DTILES_batch_table_hierarchy example "
...
+  PropertyTable& propertyTable = pExtension->propertyTables[0];
+  CHECK(propertyTable.classProperty == "default");
+
+  // Verify that all property table properties refer to a valid bufferView.
+  for (const std::pair<const std::string, PropertyTableProperty>& pair :
+       propertyTable.properties) {
+    CHECK(pair.second.values >= 0);
+    CHECK(size_t(pair.second.values) < gltf.bufferViews.size());
+  }
+}
```
Comment (j9liu) [inline Cesium3DTilesContent/test/TestUpgradeBatchTableToExtStructuralMetadata.cpp:2460]: I'd add an explicit check that "missingValues" isn't present, for sanity's sake. (I can do that)

File: CHANGES.md [hunk 3]
```diff
@@ -12,11 +12,27 @@
...
 
 - Fixed a bug in `joinToString` when given a collection containing empty strings.
 - `QuantizedMeshLoader` now creates spec-compliant glTFs from a quantized-mesh terrain tile. Previously, the generated glTF had small problems that could confuse some clients.
+- glTFs converted from a legacy batch table to a `EXT_structural_metadata` now:
+  - Add the `EXT_structural_metadata` and `EXT_mesh_features` extensions to the glTF's `extensionsUsed` list.
+  - Omit property table properties without any values at all. Previously, such property table properties would have a `values` field referring to an invalid bufferView, which is contrary to the extension's specification.
+  - Rename the `_BATCHID` attribute to `_FEATURE_ID_0` inside the `KHR_draco_mesh_compression` extension (if present), in addition to the primitive's `attributes`. Previously, meshes still Draco-compressed after the upgrade, by setting `options.decodeDraco=false`, did not have the proper attribute name.
+- glTFs converted from the 3D Tiles B3DM format now have `CESIUM_RTC` added to their `extensionsRequired` and `extensionsUsed` lists when that extension is added because the B3DM uses the `RTC_CENTER` property.
```
Comment (j9liu) [inline CHANGES.md:25]: Just some nitpicking for easier reading.

```suggestion
- glTFs converted from 3D Tiles B3DMs with the `RTC_CENTER` property will now have `CESIUM_RTC` added to their `extensionsRequired` and `extensionsUsed` lists.
```

File: CHANGES.md [hunk 4]
```diff
@@ -12,11 +12,27 @@
...
+- glTFs converted from the 3D Tiles PNTS format now:
+  - Have their `asset.version` field correctly set to `"2.0"`. Previously the version was not set, which is invalid.
+  - Add the `KHR_materials_unlit` extension to the glTF's `extensionsUsed` list when this extension is added to the glTF because the point cloud does not have normals.
```
Comment (j9liu) [inline CHANGES.md:28]:
```suggestion
  - Have the `KHR_materials_unlit` extension added to the glTF's `extensionsUsed` list when the point cloud does not have normals.
```

File: CHANGES.md [hunk 5]
```diff
@@ -12,11 +12,27 @@
...
+  - Have a default `scene`.
+  - Add the `CESIUM_RTC` extension to the glTF's `extensionsRequired` and `extensionsUsed` lists when that extension is added because the PNTS uses the `RTC_CENTER` property.
```
Comment (j9liu) [inline CHANGES.md:30]:
```suggestion
  - Have the `CESIUM_RTC` extension added to the glTF's `extensionsRequired` and `extensionsUsed` lists when the PNTS uses the `RTC_CENTER` property.
```

File: CHANGES.md [hunk 6]
```diff
@@ -12,11 +12,27 @@
...
+- When `KHR_texture_transform` is removed from a glTF at load time, the accessors and bufferViews created to hold the newly-generated texture coordinates now have their `byteOffset` properties correctly set to zero, instead of inheriting the value from the original `KHR_texture_transform`-dependent objects.
```
Comment (j9liu) [inline CHANGES.md:31]:
```suggestion
- When glTFs are loaded with `applyTextureTransform` set to true, the accessors and bufferViews created for the newly-generated texture coordinates now have their `byteOffset` set to zero. Previously, they inherited the value from the original `KHR_texture_transform`-dependent objects, which was incorrect.
```

File: CHANGES.md [hunk 7]
```diff
@@ -12,11 +12,27 @@
...
+- `bufferViews` created for indices during Draco decoding no longer have their `byteStride` property set, as this is unnecessary and disallowed by the specification.
+- `bufferViews` created for vertex attributes during Draco decoding now have their `target` property correctly set to `BufferView::Target::ARRAY_BUFFER`.
+- After a glTF has been Draco-decoded, the `KHR_draco_mesh_compression` extension is now removed from the primitives, as well as from `extensionsUsed` and `extensionsRequired`.
+- Accessors created for the position attribute in glTFs converted from quantized-mesh tiles now have their minimum and maximum values set correctly to include the positions of the vertices that form the skirt around the edge of the tile.
```
Comment (j9liu) [inline CHANGES.md:35]:
```suggestion
- For glTFs converted from quantized-mesh tiles, accessors created for the position attribute now have their minimum and maximum values set correctly to include the vertices that form the skirt around the edge of the tile.
```

**Review by j9liu — COMMENTED** [review]
Looks great @kring ! Seems like there were many correctness errors I failed to catch when writing some of this code :sweat_smile:

I had some minor comments, but I've gone ahead and pushed changes for them. I'll merge when CI passes!

**Comment by kring** [comment]
> Seems like there were many correctness errors I failed to catch when writing some of this code

Very minor things, really. It's always going to be hard to catch this sort of thing without having a proper glTF validator that we can use from the tests.

Thanks for all the fixes in this PR, they look good!

**Pull request merged**